rpc/connection.go

```diff
@@ -609,6 +609,9 @@ func (c *Connection) doReconnect(ctx context.Context, disconnectStatus Disconnec
 			// A fatal error happened.
 			*reconnectErrPtr = err
 			// short-circuit Retry
+			if err != nil {
+				c.transport.Close()
```

Comment:
Hmm maybe, I guess that could be a better fix. This was the path I found that could do it, there could be others I suppose.